.github/workflows/s3-workflow-rollout-to-scaleway.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Rollout S3 buckets (Scaleway)
+on:  # yamllint disable-line rule:truthy
+  workflow_call:
+    inputs:
+      # Scaleway information
+      scaleway-region:
+        description: Scaleway region to deploy to
+        required: true
+        type: string
+    secrets:
+      SCALEWAY_ACCESS_KEY:
+        description: Scaleway API key ID
+        required: true
+      SCALEWAY_SECRET_KEY:
+        description: Scaleway API key secret
+        required: true
+
+jobs:
+  generate-matrix:
+    name: Generate job matrices
+    runs-on: ubuntu-latest
+    outputs:
+      buckets: ${{ steps.filtered.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+            .github
+      - name: Generate Matrix
+        id: filtered
+        uses: dorny/paths-filter@v3
+        with:
+          filters: .github/s3-rollout-filters.yaml
+
+  rollout-buckets:
+    runs-on: ubuntu-latest
+    needs: [generate-matrix]
+    name: S3CMD
+    strategy:
+      matrix:
+        deployment: ${{ fromJson(needs.generate-matrix.outputs.buckets) }}
+    if: ${{ needs.generate-matrix.outputs.buckets != '[]' }}]
+    environment: ${{ matrix.deployment }}
+    steps:
+      - name: Checkout s3 folder
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+
```

Comment:
Nitpick: remove the newlines between "steps", make it obvious what belongs together.

Reply:
@djbe done, is there a reason for this or is it just code aesthetics?

Reply:
Mostly readability, what I try to have in code/configs is clear separation between "logical blocks". It's a bit dependent on the situation, I don't think there's a clear "rule". In this case we have the start/triggers, then the setup with the matrix, and then finally the workflow itself.